File: Year-2023/LeetCode-September-challenge/TheKWeakestRowsinaMatrix.cpp
#include<bits/stdc++.h>
using namespace std;

class Solution {
public:
    vector<int> kWeakestRows(vector<vector<int>>& mat, int k) {
        int n = mat[0].size();

        for(int i = 0; i < mat.size(); ++i) mat[i].push_back(i);
        vector<int> res(k);

        sort(mat.begin(), mat.end());

        for(int i = 0; i < k; ++i){
            res[i] = mat[i][n];
        }
        return res;
    }
};

int main(){
    vector<vector<int>> mat = {
                    {1,1,0,0,0},
                    {1,1,1,1,0},
                    {1,0,0,0,0},
                    {1,1,0,0,0},
                    {1,1,1,1,1}
                };
    int k = 3;
    Solution *s = new Solution();
    vector<int> res = s->kWeakestRows(mat, k);
    for(auto ch: res) cout<<ch<<" ";
    return 0;
}

File: Year-2023/LeetCode-September-challenge/TheKWeakestRowsinaMatrix.js
var kWeakestRows = function (mat, k) {
    let n = mat[0].length;

    for (let i = 0; i < mat.length; ++i) mat[i].push(i);

    mat.sort((a, b) => a - b);
    let res = new Array(k);
    for (let i = 0; i < k; ++i) res[i] = mat[i][n];
    return res;
};

let mat =
    [[1, 1, 0, 0, 0],
    [1, 1, 1, 1, 0],
    [1, 0, 0, 0, 0],
    [1, 1, 0, 0, 0],
    [1, 1, 1, 1, 1]],
    k = 3;
console.log(kWeakestRows(mat, k));